tests/test_dvc.py

```diff
@@ -341,3 +338,106 @@ def test_warn_on_dvcyaml_output_overlap(tmp_dir, mocker, mocked_dvc_repo, dvcyam
         logger.warning.assert_called_with(msg)
     else:
         logger.warning.assert_not_called()
+
+
+def test_get_dvc_stage_template_empty(tmp_dir, mocked_dvc_repo):
+    live = Live()
+    template = get_dvc_stage_template(live)
+
+    assert template == {
+        "stages": {
+            "dvclive": {
+                "cmd": "<python my_code_file.py my_args>",
+                "deps": ["<my_code_file.py>"],
+            }
+        }
+    }
+
+
+def test_get_dvc_stage_template_artifacts(tmp_dir, mocked_dvc_repo):
+    live = Live()
+    live.log_artifact("artifact.txt")
+    template = get_dvc_stage_template(live)
+
+    assert template == {
+        "stages": {
+            "dvclive": {
+                "cmd": "<python my_code_file.py my_args>",
+                "deps": ["<my_code_file.py>"],
+                "outs": ["artifact.txt"],
+            }
+        }
+    }
+
+
+def test_get_dvc_stage_template_artifacts_copy(tmp_dir, mocked_dvc_repo):
+    (tmp_dir / "artifact.txt").touch()
+    live = Live()
+    live.log_artifact("artifact.txt", copy=True)
+    template = get_dvc_stage_template(live)
+
+    assert template == {
+        "stages": {
+            "dvclive": {
+                "cmd": "<python my_code_file.py my_args>",
+                "deps": ["<my_code_file.py>"],
+                "outs": ["dvclive/artifacts/artifact.txt"],
+            }
+        }
+    }
+
+
+def test_get_dvc_stage_template_artifacts_cache_false(tmp_dir, mocked_dvc_repo):
+    live = Live()
+    live.log_artifact("artifact.txt", cache=False)
+    template = get_dvc_stage_template(live)
+
+    assert template == {
+        "stages": {
+            "dvclive": {
+                "cmd": "<python my_code_file.py my_args>",
+                "deps": ["<my_code_file.py>"],
+                "outs": [{"artifact.txt": {"cache": False}}],
+            }
+        }
+    }
+
+
+def test_get_dvc_stage_template_cache_images(tmp_dir, mocked_dvc_repo):
+    live = Live(cache_images=True)
+    img_pil = Image.new("RGB", (500, 500), (250, 250, 250))
+    live.log_image("plot.png", img_pil)
+    template = get_dvc_stage_template(live)
+
+    assert template == {
+        "stages": {
+            "dvclive": {
+                "cmd": "<python my_code_file.py my_args>",
+                "deps": ["<my_code_file.py>"],
+                "outs": ["dvclive/plots/images"],
+            }
+        }
+    }
+
+
+def test_get_dvc_stage_template_chdir(tmp_dir, mocked_dvc_repo, monkeypatch):
+    d = tmp_dir / "sub" / "dir"
+    d.mkdir(parents=True)
+    monkeypatch.chdir(d)
```

Comment:
I am confused about this test. In this case `cwd` would be `sub/dir` but we dump the paths relative to the root `.dvc`?

Reply:
Then in the code, I am not sure why we do need to get `cwd`  instead of just using repo root dir for relpath

Reply:
Got it, thanks for the close review. Dropped it in the latest push.